phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -3545,6 +3545,18 @@ public MutationState dropIndex(DropIndexStatement statement) throws SQLException
 		return dropTable(schemaName, tableName, parentTableName, PTableType.INDEX, statement.ifExists(), false, false);
     }
 
+    public MutationState dropCDC(DropCDCStatement statement) throws SQLException {
+        String schemaName = statement.getTableName().getSchemaName();
+        String cdcTableName = statement.getCdcObjName().getName();
+        String parentTableName = statement.getTableName().getTableName();
+        // Dropping the virtual CDC Table
+        dropTable(schemaName, cdcTableName, parentTableName, PTableType.CDC, statement.ifExists(), false, false);
+
+        String indexName = CDCUtil.getCDCIndexName(statement.getCdcObjName().getName());
+        // Dropping the uncovered index associated with the CDC Table
+        return dropTable(schemaName, indexName, parentTableName, PTableType.INDEX, statement.ifExists(), false, false);
```

Comment:
In create, we are catching SQLExceptions on index creation and translating them on to CDC so that user wouldn't be exposed to the index name that is internal to the CDC implementation. We should probably do something similar here.

Reply:
Updated: Handled the SQLExceptions from dropping the Index and created a CDC Wrapper Exception on it.

I think we need not worry about dropping the CDC index directly as we are creating the index with the 2 underscores in the beginning which is preventing the index to be dropped directly.
I tried doing the same but was getting parser exceptions - 

`0: jdbc:phoenix:localhost> DROP INDEX __CDC__CDCTABLE ON DATATABLE;
Error: ERROR 601 (42P00): Syntax error. Unexpected char: '_' (state=42P00,code=601)
org.apache.phoenix.exception.PhoenixParserException: ERROR 601 (42P00): Syntax error. Unexpected char: '_'
	at org.apache.phoenix.exception.PhoenixParserException.newException(PhoenixParserException.java:33)`

Reply:
> underscores in the beginning which is preventing the index to be dropped directly.

You can still drop it by putting double quotes around the index name.